test-e2e/docker_smoke_test.go

```diff
@@ -0,0 +1,144 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+const (
+	testDir = "waypoint-examples/docker/go"
+)
+
+func TestWaypointInstall(t *testing.T) {
+	t.Logf("Testing waypoint is available...")
+	wp := NewBinary(wpBinary, ".")
+	stdout, stderr, err := wp.Run("version")
+	if err != nil {
+		t.Errorf("unexpected error getting version: %s", err)
+	}
+
+	if stderr != "" {
+		t.Errorf("unexpected stderr output getting version: %s", err)
+	}
+
+	if !strings.Contains(stdout, "Waypoint v") {
+		t.Errorf("No version output detected:\n%s", stdout)
+	}
+}
+
+func TestWaypointDockerInstall(t *testing.T) {
+	wp := NewBinary(wpBinary, testDir)
+	stdout, stderr, err := wp.Run("install", "-platform=docker", "-accept-tos", fmt.Sprintf("-docker-server-image=%s", wpServerImage))
```

Comment:
this collided with my current install 😅 obviously won't be a problem in CI but for running locally, does go have a "Before"-esque hook for verifying a workspace is clean?

Reply:
Yeah! Good idea...I was thinking we might want a pre/post tests hook for these for this sort of thing. Maybe something easy is having the first Go test just be a verification that there isn't a previously running context running or something so the tests don't fail from a collision?

Reply:
Yeah that sounds like a good first step.